wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WbEditingAction.java

```diff
@@ -788,40 +782,12 @@ private EntityDocument parseJsonResponse(JsonNode entityNode) throws IOException
 	 * data.
 	 */
 	private String getCsrfToken() {
-		if (this.csrfToken == null) {
-			refreshCsrfToken();
-		}
-		return this.csrfToken;
-	}
-
-	/**
-	 * Obtains and sets a new CSRF token, whether or not there is already a
-	 * token set right now.
-	 */
-	private void refreshCsrfToken() {
-
-		this.csrfToken = fetchCsrfToken();
+		return connection.getOrFetchToken("csrf");
 		// TODO if this is null, we could try to recover here:
 		// (1) Check if we are still logged in; maybe log in again
 		// (2) If there is another error, maybe just run the operation again
```

Comment:
This comment does not look relevant anymore. I believe it should be moved to the `ApiConnection.getOrFetchToken` method. The `getCsrfToken` might also be removed now that it is just composed of one method call.

Reply:
Should be resolved.